src/app/ad-authoring/ad-authoring.state.ts

```diff
@@ -9,15 +9,173 @@ export interface AdAuthoringWorkflowState {
   readonly callToAction?: CallToActionEnum;
   readonly fileSrc?: string;
   readonly file?: File;
+  readonly AmpHtml: string;
 }
-// add a default state for the fields above
+
 @Injectable({
   providedIn: 'root',
 })
 export class AdAuthoringWorkflowStateContainer {
+  generateAMPHTML(
+    callToActionStr: string,
+    landingPageUrl: string,
+    landingPageType: string,
+    base64AssetStr: string,
+    assetFile: File
+  ) {
+    const adHtml =
+      '"<!doctype html><html amp4ads><head><meta charset=\\"utf-8\\"><meta name=\\"viewport\\" content=\\"width=device-width,minimum-scale=1\\"><meta name=\\"amp-cta-type\\" content=\\"' +
+      callToActionStr +
+      '\\"><meta name=\\"amp-cta-url\\" content=\\"' +
+      landingPageUrl +
+      '\\"><meta name=\\"amp-cta-landing-page-type\\" content=\\"' +
+      landingPageType +
+      '\\"><style amp4ads-boilerplate>body{visibility:hidden}<\\/style><script async src=\\"https:\\/\\/cdn.ampproject.org\\/amp4ads-v0.js\\"><\\/script><script async custom-element=\\"amp-video\\" src=\\"https:\\/\\/cdn.ampproject.org\\/v0\\/amp-video-0.1.js\\"><\\/script><\\/head><body><p>Hello, fake ad with srcdoc<\\/p>';
```

Comment:
IIUC, amp-video runtime js will be included even if there's no amp-video component in the file, can we make it conditional as well?